**Superstore Sales Analysis using Pandas**

Author: Hoor Soomro

Tools Used:

Python
Pandas
Kaggle Notebook

## Project Objective

The objective of this project is to analyze Superstore sales data using Python and Pandas to identify sales trends, top-performing categories, regions, states, customer segments, and products. The analysis aims to generate business insights and provide recommendations that can help management make informed decisions.


In [1]:
import os

for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))


/kaggle/input/datasets/hoorsoomro/superstore/train.csv


In [2]:

import pandas as pd

# Load dataset
df = pd.read_csv('/kaggle/input/datasets/hoorsoomro/superstore/train.csv')

# Convert dates (IMPORTANT: dayfirst=True)
df['Order Date'] = pd.to_datetime(df['Order Date'], dayfirst=True)
df['Ship Date'] = pd.to_datetime(df['Ship Date'], dayfirst=True)

# Create date columns
df['Year'] = df['Order Date'].dt.year
df['Month'] = df['Order Date'].dt.month
df['Quarter'] = df['Order Date'].dt.quarter

In [3]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Year,Month,Quarter
0,1,CA-2017-152156,2017-11-08,2017-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420.0,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2017,11,4
1,2,CA-2017-152156,2017-11-08,2017-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420.0,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,2017,11,4
2,3,CA-2017-138688,2017-06-12,2017-06-16,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036.0,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2017,6,2
3,4,US-2016-108966,2016-10-11,2016-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311.0,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,2016,10,4
4,5,US-2016-108966,2016-10-11,2016-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311.0,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2016,10,4


In [4]:
df.shape

(9800, 21)

In [5]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Year', 'Month', 'Quarter'],
      dtype='object')

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9800 entries, 0 to 9799
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Row ID         9800 non-null   int64         
 1   Order ID       9800 non-null   object        
 2   Order Date     9800 non-null   datetime64[ns]
 3   Ship Date      9800 non-null   datetime64[ns]
 4   Ship Mode      9800 non-null   object        
 5   Customer ID    9800 non-null   object        
 6   Customer Name  9800 non-null   object        
 7   Segment        9800 non-null   object        
 8   Country        9800 non-null   object        
 9   City           9800 non-null   object        
 10  State          9800 non-null   object        
 11  Postal Code    9789 non-null   float64       
 12  Region         9800 non-null   object        
 13  Product ID     9800 non-null   object        
 14  Category       9800 non-null   object        
 15  Sub-Category   9800 n

## Dataset Overview

* Dataset: Superstore Sales Dataset
* Total Records: 9,800
* Total Columns: 18

The dataset contains information about:

* Orders
* Customers
* Products
* Categories
* Regions
* States
* Sales

This project focuses on understanding business performance through exploratory data analysis.


In [7]:
df.groupby('Category')['Sales'].sum().sort_values(ascending=False)

Category
Technology         827455.8730
Furniture          728658.5757
Office Supplies    705422.3340
Name: Sales, dtype: float64

Which category generates the highest sales?
1. tech
2. furniture
3. office suplies

Technology generated the highest total sales, followed by Furniture and Office Supplies. This suggests that technology products contribute the largest share of revenue in the dataset.

In [8]:
df.groupby('State')['Sales'].sum().sort_values(ascending=False)

State
California              446306.4635
New York                306361.1470
Texas                   168572.5322
Washington              135206.8500
Pennsylvania            116276.6500
Florida                  88436.5320
Illinois                 79236.5170
Michigan                 76136.0740
Ohio                     75130.3500
Virginia                 70636.7200
North Carolina           55165.9640
Indiana                  48718.4000
Georgia                  48219.1100
Kentucky                 36458.3900
Arizona                  35272.6570
New Jersey               34610.9720
Colorado                 31841.5980
Wisconsin                31173.4300
Tennessee                30661.8730
Minnesota                29863.1500
Massachusetts            28634.4340
Delaware                 27322.9990
Maryland                 23705.5230
Rhode Island             22525.0260
Missouri                 22205.1500
Oklahoma                 19683.3900
Alabama                  19510.6400
Oregon                

California generated the highest total sales (446,306.46), making it the company's strongest-performing state. This suggests California is a key market and should remain a focus for sales and marketing efforts.

In [9]:
df.groupby('Segment')['Sales'].sum().sort_values(ascending=False)

Segment
Consumer       1.148061e+06
Corporate      6.884941e+05
Home Office    4.249822e+05
Name: Sales, dtype: float64

In [10]:
df.groupby('Product Name')['Sales'].sum().sort_values(ascending=False).head(5)

Product Name
Canon imageCLASS 2200 Advanced Copier                                          61599.824
Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind    27453.384
Cisco TelePresence System EX90 Videoconferencing Unit                          22638.480
HON 5400 Series Task Chairs for Big and Tall                                   21870.576
GBC DocuBind TL300 Electric Binding System                                     19823.479
Name: Sales, dtype: float64

In [11]:
df.groupby('Region')['Sales'].sum().sort_values(ascending=False)

Region
West       710219.6845
East       669518.7260
Central    492646.9132
South      389151.4590
Name: Sales, dtype: float64

In [12]:
df.groupby('State')['Order ID'].count().sort_values(ascending=False)

State
California              1946
New York                1097
Texas                    973
Pennsylvania             582
Washington               504
Illinois                 483
Ohio                     454
Florida                  373
Michigan                 253
North Carolina           247
Virginia                 224
Arizona                  223
Tennessee                183
Colorado                 179
Georgia                  177
Kentucky                 137
Indiana                  135
Massachusetts            135
New Jersey               122
Oregon                   122
Wisconsin                105
Maryland                 105
Delaware                  93
Minnesota                 89
Connecticut               82
Oklahoma                  66
Missouri                  66
Alabama                   61
Arkansas                  60
Rhode Island              55
Utah                      53
Mississippi               53
South Carolina            42
Louisiana                 41
Nevada  

In [13]:
df.groupby('Product Name')['Sales'].sum().sort_values(ascending=False).head(5)

Product Name
Canon imageCLASS 2200 Advanced Copier                                          61599.824
Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind    27453.384
Cisco TelePresence System EX90 Videoconferencing Unit                          22638.480
HON 5400 Series Task Chairs for Big and Tall                                   21870.576
GBC DocuBind TL300 Electric Binding System                                     19823.479
Name: Sales, dtype: float64

The Canon imageCLASS 2200 Advanced Copier generated the highest total sales, significantly outperforming all other products. The top-selling products are primarily high-value office equipment, indicating that premium business products contribute a substantial portion of the company's revenue.

💡 Business Recommendation

The company should ensure that its highest-selling products remain well-stocked and continue to receive marketing support. At the same time, management should avoid relying too heavily on a single product and should explore opportunities to increase sales of other high-performing products

In [14]:
df.groupby('State')['Order ID'].count().sort_values(ascending=False)

State
California              1946
New York                1097
Texas                    973
Pennsylvania             582
Washington               504
Illinois                 483
Ohio                     454
Florida                  373
Michigan                 253
North Carolina           247
Virginia                 224
Arizona                  223
Tennessee                183
Colorado                 179
Georgia                  177
Kentucky                 137
Indiana                  135
Massachusetts            135
New Jersey               122
Oregon                   122
Wisconsin                105
Maryland                 105
Delaware                  93
Minnesota                 89
Connecticut               82
Oklahoma                  66
Missouri                  66
Alabama                   61
Arkansas                  60
Rhode Island              55
Utah                      53
Mississippi               53
South Carolina            42
Louisiana                 41
Nevada  

#Analysis  
California not only receives the highest number of orders but also generates the highest sales. This indicates that California is both a high-demand and high-revenue market for the company.

## Conclusion

This project demonstrates how Python and Pandas can be used to answer real business questions from sales data. Through exploratory data analysis, valuable insights were generated regarding customer segments, product categories, regions, and sales performance.

The project also strengthened practical skills in Pandas, including data exploration, grouping, aggregation, sorting, and business interpretation. These findings can support data-driven decision-making and form the foundation for more advanced analysis and dashboard development.


In [15]:
df.groupby('Region')['Sales'].sum().sort_values().head(1)

Region
South    389151.459
Name: Sales, dtype: float64

In [16]:
technology = df[df['Category'] == 'Technology']

technology.groupby('Product Name')['Sales'].sum().sort_values(ascending=False).head(5)

Product Name
Canon imageCLASS 2200 Advanced Copier                        61599.824
Cisco TelePresence System EX90 Videoconferencing Unit        22638.480
Hewlett Packard LaserJet 3310 Copier                         18839.686
HP Designjet T520 Inkjet Large Format Printer - 24" Color    18374.895
Lexmark MX611dhe Monochrome Laser Printer                    16829.901
Name: Sales, dtype: float64

In [17]:
Furniture = df[df['Category'] == 'Furniture']
Furniture['Customer Name'].nunique()

705

In [18]:
df.groupby('State')['Customer Name'].nunique().sort_values(ascending=False).head(1)

State
California    570
Name: Customer Name, dtype: int64

In [19]:
total_sales= df.groupby('State')['Sales'].sum()
total_orders= df.groupby('State')['Order ID'].nunique()
avg_order= total_sales/total_orders
avg_order.sort_values(ascending=False)



State
Wyoming                 1603.136000
Vermont                 1488.228333
Rhode Island             938.542750
Missouri                 740.171667
Nevada                   727.352261
District of Columbia     716.255000
Montana                  698.669000
Indiana                  695.977143
Minnesota                678.707955
Michigan                 656.345466
Delaware                 635.418581
Kentucky                 617.938814
Virginia                 614.232348
West Virginia            604.912000
Wisconsin                599.489038
New Jersey               596.740897
Oklahoma                 578.923235
Alabama                  573.842353
New York                 560.075223
Georgia                  554.242644
Washington               532.310433
Maryland                 515.337457
Massachusetts            461.845710
North Dakota             459.955000
Louisiana                456.552500
Florida                  448.916406
California               445.415632
Arkansas              

In [20]:
df.groupby('State')['Customer ID'].nunique().sort_values(ascending=False)

State
California              570
New York                409
Texas                   367
Pennsylvania            255
Illinois                231
Washington              223
Ohio                    196
Florida                 178
North Carolina          120
Virginia                107
Michigan                105
Arizona                  99
Tennessee                84
Georgia                  80
Colorado                 74
Indiana                  67
Massachusetts            62
New Jersey               58
Kentucky                 56
Wisconsin                51
Oregon                   50
Maryland                 45
Connecticut              43
Delaware                 43
Minnesota                42
Alabama                  34
Oklahoma                 34
Missouri                 28
Arkansas                 27
Utah                     26
Mississippi              25
Rhode Island             24
Nebraska                 23
Nevada                   23
New Mexico               22
Louisiana     

In [21]:
california = df[df['State'] == 'California']

california.groupby('Segment')['Sales'].sum().sort_values(ascending=False)

Segment
Consumer       222419.0500
Corporate      144731.4105
Home Office     79156.0030
Name: Sales, dtype: float64

In [22]:
california = df[df['State'] == 'California']
california.groupby('Category')['Sales'].sum().sort_values(ascending=False)

Category
Technology         154684.1800
Furniture          152216.5355
Office Supplies    139405.7480
Name: Sales, dtype: float64

In [23]:
california = df[df['State'] == 'California']

technology = california[california['Category'] == 'Technology']

technology.groupby('Sub-Category')['Sales'].sum().sort_values(ascending=False)

Sub-Category
Phones         67139.680
Accessories    36772.900
Machines       29492.016
Copiers        21279.584
Name: Sales, dtype: float64

In [24]:
california = df[df['State'] == 'California']
california.groupby('City')['Sales'].sum().sort_values(ascending=False)

City
Los Angeles        173420.1810
San Francisco      109041.1200
San Diego           47521.0290
Fresno               7888.5285
Sacramento           7311.2780
                      ...     
Davis                  32.4000
Montebello             27.8800
Dublin                 22.0000
Antioch                19.4400
San Luis Obispo         3.6200
Name: Sales, Length: 80, dtype: float64

Los Angeles, San Francisco, and San Diego generate the majority of California's sales, indicating that the company's strongest customer base is concentrated in the state's largest metropolitan areas.

In [25]:
california = df[df['State'] == 'California']

california.groupby('Customer Name')['Sales'].sum().sort_values(ascending=False)


Customer Name
Ken Lonsdale      8341.290
Jane Waco         7381.890
Karen Ferguson    7182.766
Nick Crebassa     6275.476
Edward Hooks      5809.010
                    ...   
Ted Trevino          5.980
Meg Tillman          5.760
Phillina Ober        4.304
Stuart Calhoun       4.260
Jack Lebron          3.984
Name: Sales, Length: 570, dtype: float64

Ken Lonsdale is the highest-revenue customer in California, making him a key customer for the business. Jane Waco and Karen Ferguson are the next highest contributors, although there is a noticeable gap between the second and third highest customers.

In [ ]:
california = df[df['State'] == 'California']

california_total_sales = california['Sales'].sum()
california_total_sales

In [ ]:
california = df[df['State'] == 'California']

california_total_sales = california['Sales'].sum()
california_total_sales

In [ ]:
top_10_products = california.groupby('Product Name')['Sales'] \
                            .sum() \
                            .sort_values(ascending=False) \
                            .head(10)

top_10_products

In [ ]:
top_10_total_sales = top_10_products.sum()
top_10_total_sales

In [ ]:
top_10_percentage = (top_10_total_sales / california_total_sales) * 100
top_10_percentage

In [ ]:


# Convert Order Date to datetime
df['Order Date'] = pd.to_datetime(df['Order Date'], errors='coerce')

# Extract Year
df['Year'] = df['Order Date'].dt.year

# Group by Year and calculate total sales
yearly_sales = df.groupby('Year')['Sales'].sum().sort_values(ascending=False)

# Display result
yearly_sales

In [ ]:
df['Year'] = df['Order Date'].dt.year
df['Month']= df['Order Date'].dt.month
sales= df.groupby(['Year','Month'])['Sales'].sum().sort_values(ascending=False)
sales

In [ ]:

df['Quarter']= df['Order Date'].dt.quarter
quarter_sales= df.groupby('Quarter')['Sales'].sum().sort_values(ascending=False)
quarter_sales

Quarter 4 generates the highest sales, contributing significantly more revenue than the other quarters. Quarter 1 has the lowest sales, indicating that customer demand is strongest towards the end of the year.

In [ ]:
df['Order Date'].isna().sum()

In [ ]:
df['Shipping Days'] = df['Ship Date'] - df['Order Date']

In [ ]:
df['Shipping Days'].dtype

In [ ]:
df['Shipping Days'].mean()

In [ ]:
df.groupby('Ship Mode')['Shipping Days'].mean().sort_values()

Same Day shipping has the shortest average delivery time among all shipping modes, making it the fastest fulfillment option for customers. Standard Class has the longest average delivery time, as expected.